parse_emails/parse_emails.py

```diff
@@ -141,12 +141,20 @@ def parse(self):
             else:
 
                 raise Exception(f"Unknown file format: [{self._file_type}] for file: [{self._file_name}]")
-            output = recursive_convert_to_unicode(output)
-            self.parsed_email = output
-            return output
+            outputs = recursive_convert_to_unicode(output)
+            outputs = [remove_unicode_spaces(output) for output in outputs] if isinstance(outputs, list) else remove_unicode_spaces(outputs)
```

Comment:
weird behavior that recursive_convert_to_unicode returns list sometimes and sometimes string
somewhere we should have logic
```
if not type(output0 == list)
     output = [output]

```
Its better it was in the recursive_convert_to_unicode function itself, but if cant be done there lets do it here before calling remove_unicode_spaces

Reply:
Since it can be either dict, list, or str I think it'll be easier leaving it this way and handle the cases inside **remove_unicode_spaces**.